ffi/src/lib.rs

```diff
@@ -66,9 +126,20 @@ pub use utils::strings::{
 
 pub use utils::log;
 
+// type aliases for iterator types.
 pub type TypedValueIterator = vec::IntoIter<TypedValue>;
 pub type TypedValueListIterator = vec::IntoIter<Vec<TypedValue>>;
 
+/// A C representation of the a change provided by the transaction observers
+/// from a single transact.
+/// Holds a transaction identifier, the changes as a set of affected attributes
+/// and the length of the list of changes.
+///
+/// #Safety
+///
+/// Callers are responsible for managing the memory for the return value.
```

Comment:
Can we hyperlink in Rust docs, so that we don't have this verbosity?  Or better yet, can we make a (mostly dummy) trait that encodes this memory management style, so that it's obvious _from the code_ what the consumer is responsible for?  I.e., for types `A`, `B`, `C`, they all implement `ConsumerDestroys`, which is the `destroy` implementation...  (Maybe this is how your macro works; I will find out.)

Reply:
I was approaching this from the point of view of someone who was coming to the FFI and reading the docs wanting to implement their own native client using it. I felt that they would want to read about the safety without having to go to the code in order to figure it out and a trait implementation would obscure that. There might be a more centralized way of doing it though - maybe implementing the trait and referencing it from the crate docs. I'll noodle on it.